### ISSUE TITLE
ReferenceError: Event is not defined

### ISSUE THREAD

**Issue by Flood** [angular/angular#22902]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

SSR crashes with

```
ERROR ReferenceError: Event is not defined
          at DomEventsPlugin.patchEvent (webpack-internal:///28:3287:14)
          at new DomEventsPlugin (webpack-internal:///28:3277:15)
          at eval (ng:///AppModule/module.ngfactory.js:23:25)
          at _callFactory (webpack-internal:///0:12882:28)
          at _createProviderInstance$1 (webpack-internal:///0:12824:26)
          at resolveNgModuleDep (webpack-internal:///0:12769:17)
          at _createClass (webpack-internal:///0:12849:29)
          at _createProviderInstance$1 (webpack-internal:///0:12821:26)
          at resolveNgModuleDep (webpack-internal:///0:12769:17)
          at NgModuleRef_.get (webpack-internal:///0:14053:16)
          at resolveDep (webpack-internal:///0:14547:45)
```

## Expected behavior

It should not crash.

## Minimal reproduction of the problem with instructions
Upgraded my project (private) to Angular 5 and then it crashed. Works in Angular 4.
I think the error is in platform-browser, but I can see that you have a check `if (!Event)` there so I am not sure why it crash. But the webpack-internal:///28:3287:14 url points to that module.

## Environment
<pre><code>
Angular version: 5.2.9

For Tooling issues:
- Node version: 6.11.2  
- Platform: Windows  

Others:
Using SSR with .NET core.

**Comment by JiaLiPassion** [comment]
@Flood , could you provide a reproduce repo?

**Comment by Flood** [comment]
My repo is private but it seems to be the same issues that is described in here: https://github.com/angular/universal/issues/844#issuecomment-346929471

And here is a repo: https://github.com/wpcfan/universal-starter.git

**Comment by JiaLiPassion** [comment]
@Flood , thank you, I think this issue is caused by the code I added, I will check it.

**Comment by JiaLiPassion** [comment]
@Flood , the `patchEvent` error is caused by my code, but the https://github.com/angular/universal/issues/844#issuecomment-346929471 seems to be another issue caused by `@HostListener`. I will first resolve the `patchEvent` one.

**Comment by Flood** [comment]
Ok great, thanks! Do you know when the fix can be released?

**Comment by JiaLiPassion** [comment]
@Flood , I am not sure the release date, I will try to find out the  reason of https://github.com/angular/universal/issues/844#issuecomment-346929471  first

**Comment by natilivni** [comment]
Any update on this?

**Comment by chrisvfabio** [comment]
After many hours I managed to fix the Event undefined and document undefined errors by changing the order of imports in the app.server.module: 

```
 imports: [
    AppModule,
    NoopAnimationsModule,
    ServerModule,
    ServerTransferStateModule,
    ModuleMapLoaderModule,
  ],
```

- Importing `ServerModule` before `AppModule` was causing the Event undefined error (This is documented so my bad)
- Importing `NoopAnimationsModule` before `AppModule` was causing a document undefined error, because `AppModule` imports `BrowserAnimationsModule` and I'm assuming that was override the `NoopAnimationsModule` import. (This worked previously in v5)

Other errors I was getting:
`ERROR ReferenceError: XMLHttpRequest is not defined`. This error occurs when importing `HttpClientModule` in AppModule **and** a LazyModule. Found fix here: https://github.com/angular/universal-starter/issues/448#issuecomment-333158166

Hope this helps.
